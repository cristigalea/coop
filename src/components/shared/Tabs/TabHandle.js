import React from 'react';
import PropTypes from 'prop-types';

import { StyleSheet, css } from 'aphrodite';

const TabHandle = ({isActive = false, onClick, children}) => {
    const className = isActive ? css(styles.handle, styles.handleActive) : css(styles.handle);
    return <li onClick={onClick} className={className}>{children}</li>;
}

TabHandle.propTypes = {
    isActive: PropTypes.bool,
    onClick: PropTypes.func.isRequired,
    children: PropTypes.any.isRequired
}

export default TabHandle;

const baseColor = '#054f9b';
const accentColor = '#fff';

const styles = StyleSheet.create({
    handle: {
        flex: 1,
        fontSize: '15px',
        display: 'flex',
        justifyContent: 'center',
        alignItems: 'center',
        textAlign: 'center',
        fontWeight: 'bold',
        cursor: 'pointer',
        color: baseColor,
        borderLeft: `2px solid ${baseColor}`,
        ':first-of-type': {
            borderLeft: 0
        },
        '@media screen and (max-width: 640px)': {
            fontSize: '12px'
        },
        '@media screen and (max-width: 500px)': {
            fontSize: '10px'
        }
    },
    handleActive: {
        backgroundColor: baseColor,
        color: accentColor
    }
});