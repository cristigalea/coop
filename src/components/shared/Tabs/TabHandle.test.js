import React from 'react';
import {shallow} from 'enzyme';

import TabHandle from './TabHandle';
import { StyleSheetTestUtils} from 'aphrodite';

describe('<TabHandle />', () => {
    beforeEach(() => {
        StyleSheetTestUtils.suppressStyleInjection();
    });
      
    afterEach(() => {
        StyleSheetTestUtils.clearBufferAndResumeStyleInjection();
    });

    it('should match the snapshot for default tabs', () => {
        const component = shallow(<TabHandle onClick={() =>{}}>Some Text</TabHandle>);
        expect(component).toMatchSnapshot();
    });
    it('should match the snapshot for active tabs', () => {
        const component = shallow(<TabHandle isActive={true} onClick={() =>{}}>Some Text</TabHandle>);
        expect(component).toMatchSnapshot();
    });
});