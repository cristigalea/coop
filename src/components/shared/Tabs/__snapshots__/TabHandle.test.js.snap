// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TabHandle /> should match the snapshot for active tabs 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TabHandle
    isActive={true}
    onClick={[Function]}
  >
    Some Text
  </TabHandle>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": "Some Text",
      "className": "handle_1wb9n5m-o_O-handleActive_wzmzt7",
      "onClick": [Function],
    },
    "ref": null,
    "rendered": "Some Text",
    "type": "li",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "Some Text",
        "className": "handle_1wb9n5m-o_O-handleActive_wzmzt7",
        "onClick": [Function],
      },
      "ref": null,
      "rendered": "Some Text",
      "type": "li",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`<TabHandle /> should match the snapshot for default tabs 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TabHandle
    onClick={[Function]}
  >
    Some Text
  </TabHandle>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": "Some Text",
      "className": "handle_1wb9n5m",
      "onClick": [Function],
    },
    "ref": null,
    "rendered": "Some Text",
    "type": "li",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "Some Text",
        "className": "handle_1wb9n5m",
        "onClick": [Function],
      },
      "ref": null,
      "rendered": "Some Text",
      "type": "li",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
